### PR TITLE
[IGA] Add iga_output_process

### PR DESCRIPTION
**Pull request by tteschemacher**
Cheerio,

**Description**
Adds the output process which outputs accordingly to quadrature point geometries. It works only with single integration point geometries and is shaped to work together with our Rhino-workflow.

**Changelog**
Added
- iga_output_process.py

Thanks for your quick review! ;)

### CODE REVIEW

File: applications/IgaApplication/python_scripts/iga_output_process.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
```
Comment (dagmawibekel) [inline applications/IgaApplication/python_scripts/iga_output_process.py]: This is not needed anymore

Comment (tteschemacher) [reply applications/IgaApplication/python_scripts/iga_output_process.py]: absolutely, thanks!!

**Comment by tteschemacher** [comment]
more comments?

**Comment by philbucher** [comment]
I have some, coming soon
(Spoiler alert, test is also missing)

File: applications/IgaApplication/python_scripts/iga_output_process.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            CreateVariablesListFromInput(self.params["integration_point_results"])
+
+        # Set up output frequency and format
+        output_file_label = self.params["file_label"].GetString()
+        if output_file_label == "time":
+            self.output_label_is_time = True
+        elif output_file_label == "step":
+            self.output_label_is_time = False
+        else:
+            msg = "{0} Error: Unknown value \"{1}\" read for parameter \"{2}\"".format(self.__class__.__name__,output_file_label,"file_label")
```
Comment (philbucher) [inline applications/IgaApplication/python_scripts/iga_output_process.py]:
```suggestion
            msg = '{} Error: Unknown value "{}" read for parameter "file_label"'.format(self.__class__.__name__,output_file_label)
```

File: applications/IgaApplication/python_scripts/iga_output_process.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+def GetPrettyTime(time):
+    pretty_time = "{0:.12g}".format(time)
+    pretty_time = float(pretty_time)
+    return pretty_time
+
+def CreateVariablesListFromInput(param):
+    '''Parse a list of variables from input.'''
+    scalar_variables = []
+    vector_variables = []
+    addmissible_scalar_types = ["Bool", "Integer", "Unsigned Integer", "Double", "Component"]
```
Comment (philbucher) [inline applications/IgaApplication/python_scripts/iga_output_process.py]:
```suggestion
    admissible_scalar_types = ["Bool", "Integer", "Unsigned Integer", "Double"]
```
this no longer exists

File: applications/IgaApplication/python_scripts/iga_output_process.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    addmissible_vector_types = ["Array", "Vector"]
+
+    # Retrieve variable name from input (a string) and request the corresponding C++ object to the kernel
+    for i in range(param.size()):
+        var_name = param[i].GetString()
+        variable = KratosMultiphysics.KratosGlobals.GetVariable(var_name)
+        var_type = KratosMultiphysics.KratosGlobals.GetVariableType(var_name)
```
Comment (philbucher) [inline applications/IgaApplication/python_scripts/iga_output_process.py]: I would use the [Core function](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/python_scripts/kratos_utilities.py#L70) and then split by types:

File: applications/IgaApplication/python_scripts/iga_output_process.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    pretty_time = "{0:.12g}".format(time)
+    pretty_time = float(pretty_time)
+    return pretty_time
+
+def CreateVariablesListFromInput(param):
+    '''Parse a list of variables from input.'''
+    scalar_variables = []
+    vector_variables = []
+    addmissible_scalar_types = ["Bool", "Integer", "Unsigned Integer", "Double", "Component"]
+    addmissible_vector_types = ["Array", "Vector"]
```
Comment (philbucher) [inline applications/IgaApplication/python_scripts/iga_output_process.py]:
```suggestion
    admissible_vector_types = ["Array", "Vector"]
```

**Review by philbucher — COMMENTED** [review]
some comments, also a test is missing
=> check the test for the gid output, you could with little effort do the same here, all the necessary functionalities are thee already

File: applications/IgaApplication/python_scripts/iga_output_process.py (hunk 3 above)
Comment (tteschemacher) [reply applications/IgaApplication/python_scripts/iga_output_process.py]: Yoo

File: applications/IgaApplication/python_scripts/iga_output_process.py (hunk 4 above)
Comment (tteschemacher) [reply applications/IgaApplication/python_scripts/iga_output_process.py]: thanks for that hint. I tried that. But I have no idea how to split up afterwards :P

**Comment by tteschemacher** [comment]
> some comments, also a test is missing
> => check the test for the gid output, you could with little effort do the same here, all the necessary functionalities are thee already

Thanks a lot. I updated all code suggestions.

Tests will come at a later stage. Most probably some smaller modifications in the output might be required.

File: applications/IgaApplication/python_scripts/iga_output_process.py (hunk 4 above)
Comment (tteschemacher) [reply applications/IgaApplication/python_scripts/iga_output_process.py]: Cheers ! done

**Review by philbucher — APPROVED** [review]

**Comment by tteschemacher** [comment]
Merci beaucoup!

**Pull request merged**